src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -2570,6 +2550,13 @@ skip:
         drop_connection(context, true, false);
         return 1;
       }
+      if (!first && m_core.have_block(h))
```

Comment:
Can you clarify why the check is skipped for the first block?

Reply:
It's the block we have in common, where the new claimed chain attaches.